pygeos/test/test_creation_indices.py

```diff
@@ -179,13 +187,7 @@ def test_invalid_indices_collections(func, indices):
     [
         ([point, line_string], [0, 0], [geom_coll([point, line_string])]),
         ([point, line_string], [0, 1], [geom_coll([point]), geom_coll([line_string])]),
-        (
-            [point, line_string],
-            [1, 1],
-            [geom_coll([]), geom_coll([point, line_string])],
-        ),
         ([point, None], [0, 0], [geom_coll([point])]),
-        ([point, None], [0, 1], [geom_coll([point]), geom_coll([])]),
```

Comment:
When actually passing `None` (so it's not a missing index), it might still be unambiguous to create an empty collection? The None is also skipped when it's not the only value for a given collection (the test case on the line just above)

Reply:
I think we will make things more complex if we handle the case of "no index" and "geometry is None" differently. But let's discuss this further in #345

Reply:
That's true, but you are still handling `None` in generally, if it's not the only geometry for a given collection, so that also gives some inconsistency ..

Reply:
As described in #345, I adapted the behaviour as per your suggestion. Missing indices give errors, but indices pointing to `None` are ignored (which may lead to an EMPTY geometry creation)